File: 프로그래머스/2/42578. 의상/의상.js
function getClothes (clothes) {
    const hash = new Map();
    for (let [name, category] of clothes) {
        hash.set(category, (hash.get(category) || 0) + 1);
    }
    return hash;
}

function countCombination (hash) {
    let answer = 1;
    for (let [category, count] of hash) {
        answer *= (count + 1);
    }
    return answer - 1;
}

function solution(clothes) {
    const hash = getClothes(clothes);
    return countCombination(hash);
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    dictionary = {}
    
    # 의상 종류별로 세기
    for name, category in clothes:
        if category in dictionary:
            dictionary[category] += 1
        else:
            dictionary[category] = 1
    
    # 조합 계산
    for count in dictionary.values():
        answer *= (count + 1)
            
    return answer - 1